R/merge_altexp.R

```diff
@@ -0,0 +1,42 @@
+#' Add an alternative experiment to a SingleCellExperiment by "left join"
+#'
+#' This function takes SingleCellExperiment (SCE) and adds an Alternative Experiment (AltExp) to it.
+#' If the two experiments do not have the same set of cells, this will use the first
+#' SCE as the base set, adding columns (cells) to the AltExp so that the matrices match on that dimension.
+#' Columns (cells) that are not found in the base experiment are discarded.
+#'
+#' @param sce a SingleCellExperiment object
+#' @param alt_exp a second SummarizedExperiment to add as an alternative experiment
+#'   Most likely, this will also be a SingleCellExperiment, but it does not have to be.
+#'   The AltExp in the output will be a SingleCellExperiment
+#' @param alt_name The name to use for the alternative experiment slot
+#'
+#' @return A SingleCellExperiment with an AltExp slot containing another SingleCellExperiment.
+#' @export
+#'
+#' @import SingleCellExperiment
+#' @importFrom S4Vectors metadata
+#'
+merge_altexp <- function(sce, alt_exp, alt_name){
+  if(!is(sce, "SingleCellExperiment")){
+    stop("sce must be a SingleCellExperiment object")
+  }
+  if(!is(alt_exp, "SummarizedExperiment")){
+    stop("alt_exp must be a SummarizedExperiment (or SingleCellExperiment) object")
+  }
+  sce_cells <- colnames(sce)
+  alt_cells <- colnames(alt_exp)
+  alt_rows <- rownames(alt_exp)
```

Comment:
```suggestion
  alt_rows <- rownames(alt_exp)
  
  # check that at least 50% of cells in alt experiment are found in original sce
  if(length(intersect(sce_cells, alt_cells)) < length(alt_cells)*0.50){
    warning("Alternative experiment has less than 50% overlap with SingleCellExperiment")
  }
```
I think that it's fair to say that at least half of the cells from the alternative experiment that you are adding should also be present in the current dataset. This is based on the assumption that the primary sce is always going to contain the rna counts, which I think is a safe assumption. However, having less than that could be just a bad quality experiment too, so I wouldn't necessarily want to stop because of that, but I think it would be worth knowing that information. This may be where we would want to add a warning like this to the QC report (similar to the annoying cellranger warnings at the top of their reports). I added this in here, because I think it's a simple way of testing for it, but I'm not 100 % sold and I also think the 50% is arbitrary but in my testing of some of the CITE-seq samples, the majority of the barcodes in the alt exp were found in the original sce so I think 50% is pretty lenient.